### PR TITLE
feat(gateway): verify Telegram webhook secret token

### LINKED ISSUE #312: [L2][Telegram] Webhook verification, parser, and command normalization
## Parent
- #295

## Depends On
- #305
- #311

## Objective
Implement Telegram webhook verification, payload parsing, and command extraction.

## Scope
1. Validate Telegram webhook authenticity and token.
2. Parse update payloads and extract bot command payload.
3. Normalize Telegram command into gateway command contract.

## Acceptance Criteria
- Telegram updates translate reliably to gateway command input.
- Invalid or malformed updates are rejected safely.
- Parser and verification are unit-tested.

### PR DESCRIPTION
**Pull request by Keith-CY**
## Summary
- add `verifyTelegramWebhookSecret` to gateway provider parser utilities
- use constant-time comparison for webhook secret checks when secret validation is configured
- add parser tests covering optional secret mode, missing/invalid secret rejection, and whitespace normalization
- update provider setup runbook to include `secret_token` usage in Telegram `setWebhook` example

## Why
Issue #312 requires Telegram webhook verification plus robust command normalization and parser test coverage. Parser normalization was already present; this PR adds explicit secret-token verification and tests.

## Test Commands and Evidence
- `cd gateway && bun run check`
- `cd gateway && bun test src/providers/parsers.test.ts src/providers/parsers.e2e.test.ts`
  - Evidence: `59 pass, 0 fail`

Closes #312

### CODE REVIEW

File: gateway/src/providers/parsers.ts [hunk 1]
```diff
@@ -57,6 +58,23 @@ export function parseTelegramUpdateToCommand(payload: unknown): NormalizedGatewa
...
+  if (expected.length === 0) {
+    return true;
+  }
+
+  const provided = providedSecret?.trim() ?? "";
+  if (provided.length === 0) {
+    return false;
+  }
+
+  return constantTimeStringEquals(provided, expected);
```
Comment (dev01lay2) [inline gateway/src/providers/parsers.ts]: NBS: `constantTimeStringEquals` short-circuits on length mismatch (`leftBuffer.length !== rightBuffer.length`), which leaks length information via timing. For webhook secrets (typically fixed-length tokens) this is practically fine, but if you want full constant-time behavior, consider padding both buffers to the same length (or hashing both before comparing, like PR #1046 does for bearer auth).

**Review by dev01lay2 — APPROVED** [review]
CI green. Clean implementation — `verifyTelegramWebhookSecret` with `timingSafeEqual` is the right approach. Tests cover no-config passthrough, missing-secret rejection, exact match, whitespace trimming, and near-miss rejection. LGTM.

**Review by Akane-CN — APPROVED** [review]
NBS: Telegram webhook secret token verification is straightforward and validated by CI. Approved.
